packages/server/src/app/handlers/handle-wallet-message.ts

```diff
@@ -0,0 +1,46 @@
+import { RelayableAction } from 'magmo-wallet/lib/src/communication';
+import { MessageRelayRequested } from '../../../../magmo-wallet-client/lib';
+import { getProcess } from '../../wallet/db/queries/walletProcess';
+import { handleNewProcessAction } from './handle-new-process-action';
+import { handleOngoingProcessAction } from './handle-ongoing-process-action';
+
+export async function handleWalletMessage(
+  message: RelayableAction,
+): Promise<MessageRelayRequested | undefined> {
+  if (await isNewProcessAction(message)) {
+    return handleNewProcessAction(message);
+  } else if (await isProtocolAction(message)) {
+    return handleOngoingProcessAction(message);
+  } else {
+    return undefined;
+  }
+}
+
+function isNewProcessAction(action: RelayableAction): boolean {
```

Comment:
if this is a type guard, then `handleNewProcessAction` can accept only the `CONCLUDE_INSTIGATED` type, which simplifies the implementation of `handleNewProcessAction`